azure-pipelines.yml

```diff
@@ -91,6 +91,10 @@ jobs:
   - bash: ci/scripts/clang-format.sh $SYSTEM_PULLREQUEST_TARGETBRANCH
     condition: eq(variables['Build.Reason'], 'PullRequest')
     displayName: clang-format (C/C++ lint)
+    # Use json schema to check testplans formatting
```

Comment:
Isn't this going to run the script twice? Do we really need it in both `azure-pipelines.yml` and `quick-lint.sh`?

Reply:
From my understanding, `ci/scripts/quick-lint.sh` itself is not invoked as part of the quality (quick lint) pipeline. Instead, `quick-lint.sh` is just a script that can be used by developers to run linting locally before they PR (albeit, it has been broken for a while which I have a fix for in #24527, so it doesn't seem to be used much). Hence the comment to "keep them in sync" in this file. Please let me know if any part of my understanding is wrong here.